server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java

```diff
@@ -0,0 +1,433 @@
+import com.oceanbase.odc.service.connection.model.ConnectionConfig;
+import com.oceanbase.odc.service.flow.task.util.TaskDownloadUrlsProvider;
+import com.oceanbase.odc.service.objectstorage.cloud.CloudObjectStorageService;
+import com.oceanbase.odc.service.schedule.job.SqlPlanJobReq;
+import com.oceanbase.odc.service.session.OdcStatementCallBack;
+import com.oceanbase.odc.service.session.factory.DefaultConnectSessionFactory;
+import com.oceanbase.odc.service.session.model.SqlExecuteResult;
+import com.oceanbase.odc.service.sqlplan.model.SqlPlanTaskResult;
+import com.oceanbase.odc.service.task.caller.JobContext;
+import com.oceanbase.odc.service.task.constants.JobParametersKeyConstants;
+import com.oceanbase.odc.service.task.util.JobUtils;
+import com.oceanbase.tools.dbbrowser.parser.ParserUtil;
+import com.oceanbase.tools.dbbrowser.parser.constant.GeneralSqlType;
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class SqlPlanTask extends BaseTask<SqlPlanTaskResult> {
+
+    private SqlPlanJobReq parameters;
+
+    private ConnectionSession connectionSession;
+
+    private SyncJdbcExecutor executor;
+
+    private SqlPlanTaskResult result;
+
+    private volatile boolean canceled = false;
+
+    private File resultJsonFile;
+
+    private String zipFileRootPath;
+
+    private String errorRecordPath = null;
+
+    private final List<SqlExecuteResult> queryResultSetBuffer = new ArrayList<>();
+
+    private final List<CSVExecuteResult> csvFileMappers = new ArrayList<>();
+
+    @Override
+    protected void doInit(JobContext context) {
+        this.result = new SqlPlanTaskResult();
+        this.parameters = JobUtils.fromJson(getJobParameters().get(JobParametersKeyConstants.META_TASK_PARAMETER_JSON),
+                SqlPlanJobReq.class);
+        this.connectionSession = generateSession();
+        this.executor = connectionSession.getSyncJdbcExecutor(ConnectionSessionConstants.CONSOLE_DS_KEY);
+        initFile();
+    }
+
+    @Override
+    protected boolean doStart(JobContext context) throws Exception {
+        String sqlContent = parameters.getSqlContent();
+        log.info("sql content={}", sqlContent);
+        // todo: Download the attachment that needs to execute SQL from OSS
+        SqlCommentProcessor processor = ConnectionSessionUtil.getSqlCommentProcessor(connectionSession);
+
+        List<OffsetString> offsetStringList = processor.split(new StringBuffer(), sqlContent);
+        result.setTotalStatements(offsetStringList.size());
+
+        for (int i = 0; i <= offsetStringList.size() - 1; i++) {
+            OffsetString offsetString = offsetStringList.get(i);
+            // The retry statement will write the result into the buffer, while executing a new SQL command will
+            // clear the buffer.
+            queryResultSetBuffer.clear();
+            String sql = offsetString.getStr();
+            try {
+                boolean success = retryStatement(sql);
```

Comment:
why `retryStatement`? It seems not to be about a 'retry'.

Reply:
The retryStatement method executes the timeout sql based on the number of retries and retry interval.

Reply:
The method name may be like `executeSqlWithRetries`?

Reply:
The method name executeSqlWithRetries has been changed to avoid misunderstandings.